### PR TITLE
refactor!: remove --k8s-control-plane-url CLI flag from Helm Chart

### LINKED ISSUE #71: Removal of --k8s-control-plane-url CLI flag
With #68 we're totally relying on the configuration provided by the Service Account options (when running in in-cluster mode) or the provided `kubeconfig` file (when running in out-of-the-cluster mode).

The code actually warns users running the development branch of the deprecation policy: for 0.1.0 this has to be removed since there's any more need for it.

# Tasks

- [ ] [Update Helm chart](#105)
- [x] [Update code](#107)

### PR DESCRIPTION
**Pull request by ludusrusso**
Related to #71

### CODE REVIEW

**Review by prometherion — CHANGES_REQUESTED** [review]
We're missing the Helm Chart version bump.

**Comment by ludusrusso** [comment]
I don't remember if I need to remove the flag from the code or only from the helm chart

File: internal/options/kube.go [hunk 1]
```diff
@@ -20,11 +20,8 @@ type kubeOpts struct {
 	config    *rest.Config
 }
 
-func NewKube(controlPlaneURL string, groupName string, claimName string, config *rest.Config) (ListenerOpts, error) {
+func NewKube(groupName string, claimName string, config *rest.Config) (ListenerOpts, error) {
```
Comment (prometherion) [inline internal/options/kube.go:23]: I don't want to be pedantic but can I ask you to split changes into several commits?

One for the Helm charts, another for the Go code-base changes.

Comment (ludusrusso) [reply internal/options/kube.go:23]: Yep, but I'm not sure if we should do this!

**Comment by prometherion** [comment]
> I don't remember if I need to remove the flag from the code or only from the helm chart

Well, you raised a good topic.

I think we should create two MRs: one for the code, the second one for the Charts.

Saying this since we have to release, first, the new binary along with Docker image, otherwise, we could end up with errors for users installing in the release process.

**Comment by ludusrusso** [comment]
> > I don't remember if I need to remove the flag from the code or only from the helm chart
> 
> Well, you raised a good topic.
> 
> I think we should create two MRs: one for the code, the second one for the Charts.
> 
> Saying this since we have to release, first, the new binary along with Docker image, otherwise, we could end up with errors for users installing in the release process.

Got it so this one is only for Helm chart!

File: charts/capsule-proxy/Chart.yaml [hunk 2]
```diff
@@ -3,4 +3,4 @@ appVersion: 0.0.5
 description: A Helm chart for Kubernetes
 name: capsule-proxy
 type: application
-version: 0.1.3
+version: 0.1.4
```
Comment (ludusrusso) [inline charts/capsule-proxy/Chart.yaml:6]: @prometherion is this right?

Comment (prometherion) [reply charts/capsule-proxy/Chart.yaml:6]: LGTM

**Comment by prometherion** [comment]
@ludusrusso may I ask you to refer the `appVersion` to v0.1.0?

Once done, we can merge this!

**Comment by ludusrusso** [comment]
> @ludusrusso may I ask you to refer the `appVersion` to v0.1.0?
> 
> Once done, we can merge this!

done!

**Review by prometherion — APPROVED** [review]
LGTM!

**Pull request merged**